### ISSUE TITLE
Webcomponents v5

### ISSUE THREAD

**Issue by vbombaerts** [SPW-DIG/metawal-core-geonetwork#907]
Améliorations pour les webcomponents v5

- [ ] Pouvoir configurer une langue pour les textes
![image](https://github.com/user-attachments/assets/d32f7fef-5b74-4806-985c-2506e4413b34)
![image](https://github.com/user-attachments/assets/fc8135de-4121-4e7d-9cd1-a04e283589e0)

- [ ] Choix des colonnes https://primeng.org/table#column-toggle
- [x] Pouvoir afficher un caption plutôt que le nom de l'élément dans les zones de filtre et de recherche
![image](https://github.com/user-attachments/assets/1978fa73-0051-4d59-8c3f-a40e7fa44f9b)

- [x] Export csv

- [x] Pouvoir définir un critère de tri par défaut 

- [x] autoriser l'utilisateur à trier les résultats sur certains critères 

- [x] Pouvoir configurer une sélection multiple dans les filtres (par exemple : filtrer les ressources de type données ou série)
![image](https://github.com/user-attachments/assets/91865ea8-9e30-4955-ae3e-1e37b8e29290)

- [ ] Pouvoir ajouter des icônes dans certaines colonnes en fonction d'un autre élément de l'index
![image](https://github.com/user-attachments/assets/97912c49-19c1-450c-94c0-fe2702f57896)

**Comment by fxprunayre** [comment]
En cours
https://apps.titellus.net/webcomponents/webcomponents
* Export CSV - pour l'instant export assez brute sans formattage des objets
* Pouvoir configurer une sélection multiple dans les filtres
![image](https://github.com/user-attachments/assets/8227b83d-9a1f-4db1-8ca1-0693565954c6)
* Pouvoir définir un critère de tri par défaut
![image](https://github.com/user-attachments/assets/2b6c6bf5-ab82-494e-b725-5102b66c9b86)

**Comment by vbombaerts** [comment]
Au niveau du webcomposant tableau de données, est-ce qu'il peut se connecter à une API plutôt qu'à un csv ?

**Comment by GeryNi** [comment]
En l'état, le webcomponent utilise une URL et s'attend à recevoir un xml. Je dirais donc oui, à condition que l'API fournisse une méthode 'GET' qui retourne un fichier csv. 

Aussi non il faudra voir ce que retourne l'API et adapter en conséquence.

**Comment by fxprunayre** [comment]
> Au niveau du webcomposant tableau de données, est-ce qu'il peut se connecter à une API plutôt qu'à un csv ?

Oui, il y a un exemple avec ODS https://www.odwb.be/api/explore/v2.1/catalog/datasets/pw_agenda/exports/csv?lang=fr&timezone=Europe%2FBrussels&use_labels=true&delimiter=,
et comme indique Géry, il faut une opération dans l'API qui retourne les données avec un GET.
